contracts/Vote.sol

```diff
@@ -0,0 +1,427 @@
+     * @return totalAbstainVotes SUM of all AccountContract abstain votes for proposal.
+     */
+    function revokeVotes(address accountVoter, uint256 proposalId)
+        public
+        onlyManager
+        returns (
+            uint256 totalYesVotes,
+            uint256 totalNoVotes,
+            uint256 totalAbstainVotes
+        )
+    {
+        (, totalYesVotes, totalNoVotes, totalAbstainVotes) = voteProposal(
+            accountVoter,
+            proposalId,
+            0,
+            0,
+            0
+        );
+        return (totalYesVotes, totalNoVotes, totalAbstainVotes);
+    }
+
+    /**
+     * @notice Returns save timestamp of proposal.
+     * @param proposalId The proposal UUID.
+     * @return The timestamp of proposal.
+     */
+    function getProposalTimestamp(uint256 proposalId) public view returns (uint256) {
+        (, , uint256 timestamp, , ) = getGovernance().getProposal(proposalId);
+        return timestamp;
+    }
+
+    /**
+     * @notice Retuns currently locked celo in voting. (This celo cannot be unlocked.)
+     * And it will remove voted proposals from account history if appropriate.
+     * @param beneficiary The beneficiary.
+     */
+    function updateHistoryAndReturnLockedStCeloInVoting(address beneficiary)
+        public
+        onlyManager
+        returns (uint256 lockedAmount)
+    {
+        Voter storage voter = voters[beneficiary];
+
+        uint256 i = voter.votedProposalIds.length;
+        while (i > 0) {
+            uint256 proposalId = voter.votedProposalIds[--i];
+            uint256 proposalTimestamp = proposalTimestamps[proposalId];
+
+            if (proposalTimestamp == 0) {
+                voter.votedProposalIds[i] = voter.votedProposalIds[
+                    voter.votedProposalIds.length - 1
+                ];
+                voter.votedProposalIds.pop();
+                continue;
+            }
+
+            if (block.timestamp < proposalTimestamp + referendumDuration) {
+                VoterRecord storage voterRecord = voter.proposalVotes[proposalId];
+                lockedAmount = Math.max(
+                    lockedAmount,
+                    voterRecord.yesVotes + voterRecord.noVotes + voterRecord.abstainVotes
+                );
+            } else {
+                voter.votedProposalIds[i] = voter.votedProposalIds[
+                    voter.votedProposalIds.length - 1
+                ];
+                voter.votedProposalIds.pop();
+                delete proposalTimestamps[proposalId];
+            }
+        }
+
+        uint256 stCelo = toStakedCelo(lockedAmount);
+        emit LockedStCeloInVoting(beneficiary, stCelo);
+        return stCelo;
+    }
+
+    /**
+     * @notice Retuns proposals still in referendum stage that voter voted on.
+     * @param voter The voter.
+     * @return Proposals in referendum stage.
+     * (For up to date result call updateHistoryAndReturnLockedStCeloInVoting first)
```

Comment:
Does it makes sense to call this `StillRelevant`, and in the comment say it's for ones still in referendum, if actually getting that guarantee requires manual interaction beforehand?